tests/core/test_snapshot_evaluator.py

```diff
@@ -188,6 +189,15 @@ def increment_stage_counter(evaluator) -> None:
     evaluator.evaluate(snapshot, "2020-01-01", "2020-01-02", "2020-01-02", snapshots={})
     assert f"RuntimeStage value: {RuntimeStage.EVALUATING.value}" in capsys.readouterr().out
 
+    adapter_mock.execute.assert_has_calls(
+        [
+            call([]),
```

Comment:
why are you checking these 3 empty calls?

Reply:
Because the pre-statements are rendered as `None` and so they're filtered out, producing an empty list to be passed to `EngineAdapter.execute`. Only the `@IF` macro is rendered as non-`None` once, when we are at evaluation time.

Reply:
should probaly only check that the if is called, don't need to check all of them